--- origsrc/bluefish-2.2.0/configure.ac	2011-11-05 21:11:55.000000000 -0500
+++ src/bluefish-2.2.0/configure.ac	2011-11-27 22:00:01.542290200 -0600
@@ -91,7 +91,7 @@ if test "$CC" = "gcc" ; then
 fi
 
 case "$host_os" in
-	windows* | cygwin* | mingw*)
+	windows* | mingw*)
 		AC_DEFINE([WIN32], [1], [Define to 1 if you have a Cygwin/Windows OS.])
 		LIBS="$LIBS -lintl.dll"
 		AC_MSG_NOTICE([platform is... Windows/Cygwin])
@@ -103,7 +103,12 @@ case "$host_os" in
 esac
 
 case "$host" in
-	*-*-mingw* | *cygwin* | *windows*)
+	*-*-cygwin*)
+		platform_win32=yes
+		native_win32=no
+	;;		
+	*-*-mingw* | *windows*)
+		platform_win32=yes
 		native_win32=yes
 	;;		
 	*-apple-darwin*)
@@ -111,6 +116,7 @@ case "$host" in
 		AC_MSG_NOTICE([platform is... MacOSX])
 	;;
 	*)
+		platform_win32=no
 		native_win32=no
 	;;
 esac
@@ -123,6 +129,7 @@ if test "x$native_win32" = "xyes"; then
 	AC_PROG_SED
 	AC_SUBST([SPLIT_VERSION], [`echo $VERSION | $SED -e 's/-.*//' | $SED -e 's/\./,/g'`])
 fi
+AM_CONDITIONAL([PLATFORM_WIN32], [test "x$platform_win32" = "xyes"])
 AM_CONDITIONAL([OS_WIN32], [test "x$native_win32" = "xyes"])
 
 dnl ************************
--- origsrc/bluefish-2.2.0/src/Makefile.am	2011-09-29 20:11:30.000000000 -0500
+++ src/bluefish-2.2.0/src/Makefile.am	2011-11-27 22:00:23.204529200 -0600
@@ -2,6 +2,7 @@
 AUTOMAKE_OPTIONS = std-options
 
 SUBDIRS = pixmaps\
+	. \
 	plugin_about\
 	plugin_entities\
 	plugin_htmlbar\
@@ -21,8 +22,11 @@ MAINTAINERCLEANFILES = Makefile.in confi
 
 DEFS += -DLOCALEDIR=\"$(localedir)\" -DPKGDATADIR=\"$(pkgdatadir)\" -DPKGLIBDIR=\"$(pkglibdir)\"
 
+if PLATFORM_WIN32
+AM_LDFLAGS = -Wl,--export-all-symbols,--out-implib,bluefish.exe.a
+endif
+
 if OS_WIN32
-AM_LDFLAGS = -export-all-symbols
 DEFS += -DEXE_EXPORT_SYMBOLS
 LDADD = -lws2_32
 LDADD += bluefish_rc.o
--- origsrc/bluefish-2.2.0/src/plugin_about/Makefile.am	2010-01-03 20:11:36.000000000 -0600
+++ src/bluefish-2.2.0/src/plugin_about/Makefile.am	2011-11-27 22:00:01.547290400 -0600
@@ -21,8 +21,8 @@ about_la_SOURCES = \
 
 about_la_LDFLAGS = -module -avoid-version
 
-if OS_WIN32
-about_la_LDFLAGS += -no-undefined -export-symbols $(srcdir)/about.def
+if PLATFORM_WIN32
+about_la_LDFLAGS += -no-undefined -Wl,../bluefish.exe.a
 endif
 
 about_rev.h: about_rev.h.tmp
--- origsrc/bluefish-2.2.0/src/plugin_charmap/Makefile.am	2009-11-19 20:11:21.000000000 -0600
+++ src/bluefish-2.2.0/src/plugin_charmap/Makefile.am	2011-11-27 22:00:01.549290600 -0600
@@ -21,8 +21,8 @@ charmap_la_CFLAGS = $(CHARMAP_PLUGIN_CFL
 charmap_la_LIBADD = $(CHARMAP_PLUGIN_LIBS)
 charmap_la_LDFLAGS = -module -avoid-version
 
-if OS_WIN32
-charmap_la_LDFLAGS += -no-undefined -export-symbols $(srcdir)/charmap.def
+if PLATFORM_WIN32
+charmap_la_LDFLAGS += -no-undefined -Wl,../bluefish.exe.a
 endif
 
 maintainer-clean-local:
--- origsrc/bluefish-2.2.0/src/plugin_entities/Makefile.am	2009-11-19 20:11:21.000000000 -0600
+++ src/bluefish-2.2.0/src/plugin_entities/Makefile.am	2011-11-27 22:00:01.551290700 -0600
@@ -16,8 +16,8 @@ entities_la_SOURCES = \
 
 entities_la_LDFLAGS = -module -avoid-version
 
-if OS_WIN32
-entities_la_LDFLAGS += -no-undefined -export-symbols $(srcdir)/entities.def
+if PLATFORM_WIN32
+entities_la_LDFLAGS += -no-undefined -Wl,../bluefish.exe.a
 endif
 
 maintainer-clean-local:
--- origsrc/bluefish-2.2.0/src/plugin_htmlbar/Makefile.am	2011-03-12 20:11:51.000000000 -0600
+++ src/bluefish-2.2.0/src/plugin_htmlbar/Makefile.am	2011-11-27 22:00:01.552290700 -0600
@@ -41,8 +41,8 @@ htmlbar_la_SOURCES = \
 
 htmlbar_la_LDFLAGS = -module -avoid-version
 
-if OS_WIN32
-htmlbar_la_LDFLAGS += -no-undefined -export-symbols $(srcdir)/htmlbar.def
+if PLATFORM_WIN32
+htmlbar_la_LDFLAGS += -no-undefined -Wl,../bluefish.exe.a
 endif
 
 maintainer-clean-local:
--- origsrc/bluefish-2.2.0/src/plugin_infbrowser/Makefile.am	2009-11-19 20:11:22.000000000 -0600
+++ src/bluefish-2.2.0/src/plugin_infbrowser/Makefile.am	2011-11-27 22:00:01.555290900 -0600
@@ -30,8 +30,8 @@ infbrowser_la_SOURCES = \
 
 infbrowser_la_LDFLAGS = -module -avoid-version
 
-if OS_WIN32
-infbrowser_la_LDFLAGS += -no-undefined -export-symbols $(srcdir)/infbrowser.def
+if PLATFORM_WIN32
+infbrowser_la_LDFLAGS += -no-undefined -Wl,../bluefish.exe.a
 endif
 
 maintainer-clean-local:
--- origsrc/bluefish-2.2.0/src/plugin_snippets/Makefile.am	2010-01-10 20:11:29.000000000 -0600
+++ src/bluefish-2.2.0/src/plugin_snippets/Makefile.am	2011-11-27 22:00:01.557291000 -0600
@@ -28,8 +28,8 @@ snippets_la_SOURCES = \
 
 snippets_la_LDFLAGS = -module -avoid-version
 
-if OS_WIN32
-snippets_la_LDFLAGS += -no-undefined -export-symbols $(srcdir)/snippets.def
+if PLATFORM_WIN32
+snippets_la_LDFLAGS += -no-undefined -Wl,../bluefish.exe.a
 endif
 
 maintainer-clean-local:
--- origsrc/bluefish-2.2.3/src/plugin_zencoding/Makefile.am	2011-12-05 20:11:28.000000000 -0600
+++ src/bluefish-2.2.3/src/plugin_zencoding/Makefile.am	2012-07-02 08:30:55.439235800 -0500
@@ -58,8 +58,8 @@ zencoding_la_SOURCES = \
 
 zencoding_la_LDFLAGS = -module -avoid-version
 
-if OS_WIN32
-zencoding_la_LDFLAGS += -no-undefined -export-symbols $(srcdir)/zencoding.def
+if PLATFORM_WIN32
+zencoding_la_LDFLAGS += -no-undefined -Wl,../bluefish.exe.a
 endif
 
 maintainer-clean-local:
